Introduce --upto, optional config keys, custom migration schema name, .DS_Store typo fix

pieterjandebruyne opened the pull request:
- introduce --upto command (fixed the logic so that when you use this to pass a specific migration name, that migration iwll be included
- Make configuration keys optional (I think this is better so that updates to default will also be implemented for users that did not want to override the default)
- allow custom migration schema name (older configuration files should still work because all keys are now optional and fallback to the default)
- fix ".DS_STORE" typo to ".DS_Store"
- update readme to reflect above changes

Commit:
bump packages and update .gitignore

Commit:
Make configuration keys optional, introduce --upto command, allow custom migration schema name, fix DS_STORE typo, update readme

Softjey commented on src/config/getConfig.ts:
```diff
@@ -4,12 +4,28 @@ import { ConfigSchema } from "./config.type";
+        console.info(`No config file ${CONFIG_FILE_NAME} found, using defaults`)
+      }else {
+        throw new Error(
+          "Could not read config file"
```
Could you also show the initial error to the user? This will help avoid situations where the user doesn't know the cause of the error. You could try to use throw new Error("Could not read config file", { cause: error }); for this.

pieterjandebruyne replied:
not supported in your target es2018... so none of the error currently have this. I would suggest a targetbump and align all errors if you want this. But would do it in separate PR...

Softjey commented:
Thanks for the great work! In the evening, I'll try to pull the branch locally, check if everything is fine, and publish a new version on NPM

Softjey commented:
Sorry for the delay.  
I pulled the repository locally and noticed one more small thing. It would be better to display message about the missing config file only once, instead of doing it every time it's needed.

Other than that, everything looks okay. Let me know once you've resolved these two issues, or let me know if you'd like me to fix them, and I'll publish the new version on NPM.  

Once again, thank you for your contribution!

Commit:
Add configLoader singleton

pieterjandebruyne commented:
Updated the PR,
I refactored the getConfig to be a singleton of a class so we only fetch / parse the file once.
- If we don't find a file, post info message and use default
- if we found a file but it is invalid (either json or field validation fails) we throw and exit

I did not implement the cause inside the throw new Error since we don't do it anywhere and it would require your tsconfig target to be >es2022 since that function overload is only available since that version... 

Feel free to make any changes to the logic I added if you would like to have something changed..

If you are fine with it, please release it on npm :) thx.

Softjey commented:
Nice work! Thank you!
